aequilibrae/paths/linear_approximation.py

```diff
@@ -124,30 +124,21 @@ def __init__(self, assig_spec, algorithm, project=None) -> None:
         self.previous_step_direction = {}  # type: Dict[AssignmentResults]
         self.pre_previous_step_direction = {}  # type: Dict[AssignmentResults]
```

Comment:
Can we rename `pre_previous_step_direction` to `temp_step_direction_for_copy` or something like that?  Actually can we also make it local to the bfw step direction iteration? It's only used there to hold the step_direction because we operate on it in place but need it in the next iteration as previous_step_direction

Reply:
can make this a separate PR too, but it fits with the small bug fix

Reply:
Just renamed those as well as the others I added.